### DISCUSSION TITLE
press [esc] to unselect everything

### DISCUSSION THREAD

**Discussion by poyribn** [Orama-Interactive/Pixelorama#838] [Ideas]
one thing i often do, but become disappointed because it doesn't work, is pressing [Esc] to unselect the selection that i made with the _Rectangular Selection_ tool and/or the _Elliptical Selection_ tool and/or the _Polygonal Selection_ tool and/or the _Select by Color_ tool and/or the _Magic Wand_ tool and/or the _Lasso / Free Select Tool_ tool.

i hope that pressing [Esc] to unselect everything gets added soon, because it is the most natural way to me.

**Comment by Variable-ind** [comment]
Hello there already exists a shortcut (Ctrl+ d)
You can change it to [Esc] in the preferences

**Reply by poyribn** [reply]
i looked through the shortcut settings before opening this discussion.

i searched all categories where i would expect 'unselect everything' to be:
1. Tools
2. Tool Modifiers
3. Canvas

i couldn't find where to change 'unselect everything'.

**Reply by Variable-ind** [reply]
It's in menu shortcuts (because clear selection is part of selection menu)
![Screenshot from 2023-03-19 18-15-08](https://user-images.githubusercontent.com/77773850/226177605-4295ce71-b988-40c2-a90a-c8dfca1a4b3a.png)

**Reply by poyribn** [reply]
Thank You very much.

this problem was solved for me, but i think we should still make this changed shortcut a default, because a lot of picture editing programs use [esc] for 'unselect everything'. on my computer such programs are (i confirmed):
- GIMP
- Inkscape
- LibreOffice Draw
- Paint
- Paint 3D
- there are probably a lot of others which are not on my computer, list some if you know of them

**Reply by OverloadedOrama** [reply]
Making Escape the default shortcut for this may not work well, due to Escape already being used to cancel a transformation. For example, if you started moving a part of your image with a selection and it hasn't yet been confirmed (appears semi-transparent), pressing Escape will paste it back to its original position.

**Reply by poyribn** [reply]
i don't know how difficult or worth it is, but we could add a special case if 'cancel moving the selection' and 'unselect everything' are on the same key:
- if the selection is moved and the movement is not confirmed or canceled, pressing the key will reset the movement and not unselect anything.
- if the selection has not been moved or the movement is confirmed or canceled, pressing the key will unselect everything.
